src/irreps/dnirrep.jl

```diff
@@ -0,0 +1,279 @@
+"""
+    struct DNIrrep{N} <: AbstractIrrep{Dihedral{N}}
+    DNIrrep{N}(n::Integer, isodd::Bool=false)
+    Irrep[Dihedral{N}](n::Integer, isodd::Bool=false)
+
+Represents irreps of the dihedral group ``D_N = Z_N ⋊ C`` (``Z_N`` and charge conjugation or reflection).
+
+## Properties
+
+- `j::Int`: the value of the ``Z_N`` charge.
+- `isodd::Bool`: the representation of charge conjugation.
+
+Combined these take the values ``+0, -0, 1, ..., (N - 1) / 2`` for odd ``N``, and
+``+0, -0, 1, ..., N / 2 - 1, +(N/2), -(N/2)`` for even ``N``.
```

Comment:
Not sure if this needs to be even more explicit about the relation between the signs and the `isodd` boolean, as well as about the dimensionality of the representations.